src/Microsoft.AspNetCore.Localization/CookieRequestCultureProvider.cs

```diff
@@ -36,6 +42,8 @@ public override Task<ProviderCultureResult> DetermineProviderCultureResult(HttpC
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
+            _logger = httpContext.RequestServices.GetService<ILogger>() ?? NullLogger.Instance;
```

Comment:
@pranavkm at the first glance I was thinking to inject the `ILogger` into the constructor, but while the `ParseCookieValue` is static I come up with this idea